web/src/builder.rs

```diff
@@ -0,0 +1,631 @@
+            "typewriter" => DefaultFont::Typewriter,
+            _ => {
+                return;
+            }
+        };
+        self.default_fonts.insert(
+            default,
+            fonts
+                .into_iter()
+                .flat_map(|value| value.as_string())
+                .collect(),
+        );
+    }
+
+    // TODO: This should be split into two methods that either load url or load data
+    // Right now, that's done immediately afterwards in TS
+    pub async fn build(&self, parent: HtmlElement, js_player: JavascriptPlayer) -> Promise {
+        let copy = self.clone();
+        wasm_bindgen_futures::future_to_promise(async move {
+            if RUFFLE_GLOBAL_PANIC.is_completed() {
+                // If an actual panic happened, then we can't trust the state it left us in.
+                // Prevent future players from loading so that they can inform the user about the error.
+                return Err("Ruffle is panicking!".into());
+            }
+
+            let ruffle = RuffleHandle::new_internal(parent, js_player, copy)
+                .await
+                .map_err(|err| JsValue::from(format!("Error creating player: {}", err)))?;
+            Ok(JsValue::from(ruffle))
+        })
+    }
+}
+
+impl RuffleInstanceBuilder {
+    pub fn setup_fonts(&self, player: &mut Player) {
+        for (font_name, bytes) in &self.custom_fonts {
+            if let Ok(swf_stream) = swf::decompress_swf(&bytes[..]) {
+                if let Ok(swf) = swf::parse_swf(&swf_stream) {
+                    let encoding = swf::SwfStr::encoding_for_version(swf.header.version());
+                    for tag in swf.tags {
+                        match tag {
+                            swf::Tag::DefineFont(_font) => {
+                                tracing::warn!("DefineFont1 tag is not yet supported by Ruffle, inside font swf {font_name}");
+                            }
+                            swf::Tag::DefineFont2(font) => {
+                                tracing::debug!(
+                                    "Loaded font {} from font swf {font_name}",
+                                    font.name.to_str_lossy(encoding)
+                                );
+                                player
+                                    .register_device_font(FontDefinition::SwfTag(*font, encoding));
+                            }
+                            swf::Tag::DefineFont4(font) => {
+                                let name = font.name.to_str_lossy(encoding);
+                                if let Some(data) = font.data {
+                                    tracing::debug!("Loaded font {name} from font swf {font_name}");
+                                    player.register_device_font(FontDefinition::FontFile {
+                                        name: name.to_string(),
+                                        is_bold: font.is_bold,
+                                        is_italic: font.is_bold,
```

Comment:
Shouldn't this be `font.is_italic` instead? (Though this seems to have existed before)

Reply:
Nice catch!